### PR TITLE
[merged] Refactor containers verb

### PR DESCRIPTION
**Pull request by baude**
With the exception of fstrim, the containers verb has now been
refactored.  It primarily now uses the containers object in its
implementation.

### CODE REVIEW

File: Atomic/backends/_ostree.py [hunk 1]
```diff
@@ -26,18 +26,20 @@ def backend(self):
...
-
         container = Container(container_id, backend=self)
         container.name = container_id
+        container.command = info['Command']
         container.id = container_id
+        container.image_name = info['Image']
+        container.image_id = info['ImageID']
         container.created = info['Created']
-        container.status = runtime['status']
+        container.status = container.state = runtime['status']
```
Comment (rhatdan) [inline Atomic/backends/_ostree.py]: Why do we have container.status and container.state?

File: Atomic/backendutils.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def get_backend_and_image(self, img, str_preferred_backend=None):
         raise ValueError("Found {} in multiple storage backends: {}".
                          format(img, ', '.join([x.backend for x in img_in_backends])))
 
-    def get_backend_for_container(self, container, str_preferred_backend=None):
+    def get_backend_and_container(self, container, str_preferred_backend=None):
```
Comment (rhatdan) [inline Atomic/backendutils.py]: Confusing name of method.  You are passing in "container" and returning con_obj.  Either
container should be renamed to container_id, or function should be get_backend_and_container_object

File: Atomic/backendutils.py [hunk 3]
```diff
@@ -75,16 +75,18 @@ def get_backend_for_container(self, container, str_preferred_backend=None):
...
         container_in_backends = []
         for backend in backends:
             be = backend()
-            if be.has_container(container):
-                container_in_backends.append(be)
+            con_obj = be.has_container(container)
+            if con_obj:
+                container_in_backends.append((be, con_obj))
         if len(container_in_backends) == 1:
             return container_in_backends[0]
```
Comment (rhatdan) [inline Atomic/backendutils.py]: Is this multiple objects?

File: Atomic/containers.py [hunk 4]
```diff
@@ -66,6 +68,9 @@ def cli(subparser):
 
 class Containers(Atomic):
 
+    FILTERALES= {"container": "id", "image": "image_name", "command": "command",
```
Comment (rhatdan) [inline Atomic/containers.py]: FILTERALES?  Bad name.

File: Atomic/backends/_ostree.py (hunk 1 above)
Comment (baude) [inline Atomic/backends/_ostree.py]: Depending on how the struct information is gather (i.e. inspect vs containers), they have State and/or Status and they differ.  In ostree, we set both to be the same and handle it.

File: Atomic/backendutils.py (hunk 2 above)
Comment (baude) [inline Atomic/backendutils.py]: Done

File: Atomic/containers.py (hunk 4 above)
Comment (baude) [inline Atomic/containers.py]: Fixed, although that was the previous name as well.  Now called FILTER_KEYWORDS

File: Atomic/backendutils.py (hunk 3 above)
Comment (rhatdan) [inline Atomic/backendutils.py]: Function returns two objects.  This seems to be returning 1?

Comment (baude) [inline Atomic/backendutils.py]: No, it returns an instance of the backend and a container object seen there on line 89.

**Comment by rhatdan** [comment]
If tests pass, I will merge.

**Comment by baude** [comment]
@rhatdan looks like i got it this time.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**